packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,160 @@
+| children           | `React.ReactNode`                                                        | `undefined` | Trigger component for tooltip, Accepts any interactive element or icons                                                                                  | ✅       |
+| placement          | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip                                                                                                                                     |          |
+| shouldWrapChildren | `boolean`                                                                | `false`     | If set to true, the tooltip trigger will be wrapped in a interactive `<div>`, this is useful for making non-interactive elements work as tooltip trigger |          |
+| onOpen             | `Callback`                                                               | `undefined` | Called when tooltip is opened                                                                                                                            |          |
+| onClose            | `Callback`                                                               | `undefined` | Called when tooltip is closed                                                                                                                            |          |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic example
+<Tooltip content="Amount reversed to customer bank account" placement="top">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>
+```
+
+### Working with non-interactive triggers
+
+To make tooltip apear on hovering over non-interactive elements such as icons, badges, counters etc. We will provide `shouldWrapChildren` prop which will wrap the trigger component in a interactive div element with all the trigger props
+
+```js
+// non-interactive element as trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
+  <Counter value={100} />
+</Tooltip>
+```
+
+> Note: If users don't wrap the non-interactive elements the tooltip won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
+
+To make custom elements work the components needs to expose:
+
+- ref
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+```jsx
+import type { TooltipTriggerProps } from "@razorpay/blade/components";
+
+type MyCustomButtonProps = {} & TooltipTriggerProps
+const MyCustomButton = React.forwardRef(...);
+
+// custom trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
+  <MyCustomButton>Click me<MyCustomButton>
+</Tooltip>
+```
+
+## Library
+
+We will be using [FloatingUI](https://floating-ui.com/) to position the tooltip & handle the basic tooltip logic. Floating UI supports both Web & ReactNative.
```

Comment:
Offtopic: Nice!! Will we also be able to use this DropdownOverlay?

Reply:
Yes. most certainly, and will fix also handle all those placement bugs.